packages/node-core/src/indexer/project.service.ts

```diff
@@ -370,8 +367,10 @@ export abstract class BaseProjectService<
   }
 
   private async initUpgradeService(): Promise<number | undefined> {
```

Comment:
For clarity rather than using `this._startHeight` we should use an argument here. This would change from a runtime error to a compiler error and be caught earlier.